packages/core/src/services/evaluationResults/computeEvaluationResultsByDocumentContent.ts

```diff
@@ -1,20 +1,79 @@
-import { and, desc, eq, or } from 'drizzle-orm'
+import { and, desc, eq, or, sql } from 'drizzle-orm'
 
-import { Commit, Evaluation } from '../../browser'
+import { Commit, Evaluation, Workspace } from '../../browser'
 import { database } from '../../client'
-import { hashContent, paginateQuery, Result } from '../../lib'
-import { DocumentVersionsRepository } from '../../repositories'
+import { hashContent, Result } from '../../lib'
+import { calculateOffset } from '../../lib/pagination/calculateOffset'
+import {
+  DocumentLogsRepository,
+  DocumentVersionsRepository,
+  EvaluationResultsRepository,
+} from '../../repositories'
 import { commits } from '../../schema'
 import { getResolvedContent } from '../documents'
-import { createEvaluationResultQuery } from './_createEvaluationResultQuery'
+
+function getRepositoryScopes(workspaceId: number, db = database) {
+  const evaluationResultsScope = new EvaluationResultsRepository(
+    workspaceId,
+    db,
+  ).scope
+  const documentLogsScope = new DocumentLogsRepository(workspaceId, db).scope
+  return { evaluationResultsScope, documentLogsScope }
+}
+type Scope = ReturnType<typeof getRepositoryScopes>
+
+function getCommonQueryConditions({
+  evaluationResultsScope,
+  documentLogsScope,
+  evaluation,
+  resolvedContent,
+  commit,
+}: {
+  evaluationResultsScope: Scope['evaluationResultsScope']
+  documentLogsScope: Scope['documentLogsScope']
+  evaluation: Evaluation
+  resolvedContent: string
+  commit: Commit
+}) {
+  return and(
+    eq(evaluationResultsScope.evaluationId, evaluation.id),
+    or(
+      eq(documentLogsScope.contentHash, hashContent(resolvedContent)),
+      eq(commits.id, commit.id),
+    ),
+  )
+}
```

Comment:
☝️ Review query, please. Now it does not include aggregations because the data in this table is like this:
![image](https://github.com/user-attachments/assets/0dc95141-605d-4b69-bc7c-3572e50dd8a4)